subsys/bluetooth/host/adv.c

```diff
@@ -462,20 +492,30 @@ static int hci_set_adv_ext_fragmented(struct bt_le_ext_adv *adv, uint16_t hci_op
 
 			/* We can't necessarily set one AD field in a single step. */
 			while (offset < data[j].data_len) {
-				buf = bt_hci_cmd_create(hci_op, sizeof(*set_data));
+				BUILD_ASSERT(sizeof(struct bt_hci_cp_le_set_per_adv_data)
+					     == UINT8_MAX);
+				BUILD_ASSERT(sizeof(struct bt_hci_cp_le_set_ext_adv_data)
+					     == UINT8_MAX);
+				BUILD_ASSERT(sizeof(struct bt_hci_cp_le_set_ext_scan_rsp_data)
+					     == UINT8_MAX);
```

Comment:
These asserts may conflict with what is being requested here: https://github.com/zephyrproject-rtos/zephyr/issues/37710

Reply:
I'm not sure if I follow. The size of these commands are defined by the spec. If we want to optimize the communication between cores, then we should rather modify the length we pass to `net_buf_add()`

Reply:
The point is that 255 is the maximum size of the commands, and that always sending 255 octets (which we currently do) even if we just set a single octet of data is inefficient. Of course we could keep the structs at 255 octets and, as you suggest, simply modify the number of octets we add to the buffer.

Reply:
Yeah. There are multiple ways of solving it. Lets wait for some more input on this